ShopperInsights/src/main/java/com/braintreepayments/api/EligiblePaymentsApi.kt

```diff
@@ -1,28 +1,40 @@
 package com.braintreepayments.api
 
 import com.braintreepayments.api.EligiblePaymentsApiRequest.Companion.toJson
+import org.json.JSONException
 
-internal class EligiblePaymentsApi {
-    fun execute(request: EligiblePaymentsApiRequest): EligiblePaymentsApiResult {
-        request.toJson()
-        // TODO: Network call
-
-        // Hardcoded result
-        return EligiblePaymentsApiResult(
-            eligibleMethods = EligiblePaymentMethods(
-                paypal = EligiblePaymentMethodDetails(
-                    canBeVaulted = true,
-                    eligibleInPayPalNetwork = true,
-                    recommended = true,
-                    recommendedPriority = 1
-                ),
-                venmo = EligiblePaymentMethodDetails(
-                    canBeVaulted = true,
-                    eligibleInPayPalNetwork = true,
-                    recommended = true,
-                    recommendedPriority = 1
-                )
+internal class EligiblePaymentsApi(
+    private val braintreeClient: BraintreeClient
+) {
+    fun execute(request: EligiblePaymentsApiRequest, callback: EligiblePaymentsCallback) {
+        val jsonBody = request.toJson()
+        braintreeClient.getConfiguration { configuration, configError ->
+            // TODO: Move url to PaypalHttpClient class when it is created
+            val baseUrl = when (configuration?.environment) {
+                "production" -> "https://api.paypal.com"
+                else -> "https://api.sandbox.paypal.com"
+            }
```

Comment:
Does this baseUrl logic exist in the other API calls? I wonder if we can extract this to a shared function/class.

Reply:
One exists for Braintree but not PayPal. https://github.com/braintree/braintree_android/blob/8c921affcfcfcf64f2fbce2c9c0787cbe6358595/BraintreeCore/src/main/java/com/braintreepayments/api/TokenizationKey.kt#L35

I think the plan is to create a `PaypalHttpClient` and have the baseUrl logic live in there.